templates/addons/cluster-autoscaler/workload-rbac.yaml

```diff
@@ -0,0 +1,151 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ${CLUSTER_NAME}-cluster-autoscaler-addon
+data:
+  cluster-autoscaler-workload-rbac.yaml: |
+    ---
+    kind: ClusterRoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: cluster-autoscaler
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: cluster-autoscaler
+    subjects:
+    - kind: ServiceAccount
+      name: cluster-autoscaler
+      namespace: kube-system
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: cluster-autoscaler
+      namespace: kube-system
+    ---
+    kind: ClusterRole
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: cluster-autoscaler
+    rules:
+      - apiGroups:
+        - ""
+        resources:
+        - namespaces
+        - persistentvolumeclaims
+        - persistentvolumes
+        - pods
+        - replicationcontrollers
+        - services
+        verbs:
+        - get
+        - list
+        - watch
+      - apiGroups:
+        - ""
+        resources:
+        - nodes
+        verbs:
+        - get
+        - list
+        - update
+        - watch
+      - apiGroups:
+        - ""
+        resources:
+        - pods/eviction
+        verbs:
+        - create
+      - apiGroups:
+        - policy
+        resources:
+        - poddisruptionbudgets
+        verbs:
+        - list
+        - watch
+      - apiGroups:
+        - storage.k8s.io
+        resources:
+        - csinodes
+        - storageclasses
+        - csidrivers
+        - csistoragecapacities
+        verbs:
+        - get
+        - list
+        - watch
+      - apiGroups:
+        - batch
+        resources:
+        - jobs
+        verbs:
+        - list
+        - watch
+      - apiGroups:
+        - apps
+        resources:
+        - daemonsets
+        - replicasets
+        - statefulsets
+        verbs:
+        - list
+        - watch
+      - apiGroups:
+        - ""
+        resources:
+        - events
+        verbs:
+        - create
+        - patch
+      - apiGroups:
+        - ""
+        resources:
+        - configmaps
+        verbs:
+        - create
+        - delete
+        - get
+        - update
+      - apiGroups:
+        - coordination.k8s.io
+        resources:
+        - leases
+        verbs:
+        - create
+        - get
+        - update
+---
+apiVersion: v1
+kind: Secret
+type: addons.cluster.x-k8s.io/resource-set
+metadata:
+  name: ${CLUSTER_NAME}-cluster-autoscaler-addon
+stringData:
+  cluster-autoscaler-workload-token.yaml: |-
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: cluster-autoscaler-token
+      namespace: kube-system
+      annotations:
+        kubernetes.io/service-account.name: cluster-autoscaler
+    type: kubernetes.io/service-account-token
```

Comment:
is this secret auto-generated when it is applied?

Reply:
It's something I'm testing to retrieve the token for the `cluster-autoscaler` Service Account on the workload cluster after I've created it: https://kubernetes.io/docs/concepts/configuration/secret/#serviceaccount-token-secrets